apiClientDotNet/Utils/RestRequestHandler.cs

```diff
@@ -113,48 +95,105 @@ public HttpWebResponse executeRequest(object data, String url, bool isAuth, stri
                     response = we.Response as HttpWebResponse;
                     Console.Write(response);
                 }
+
                 response = null;
             }
+
             return response;
         }
 
+        private void SetProxy(HttpWebRequest request, SymConfig symConfig)
+        {
+            string proxyUrl = null;
+            string proxyUserName = null;
+            string proxyPassword = null;
+
+            // Find proxy
+            var url = request.RequestUri.AbsolutePath;
+            if (url.Contains(symConfig.sessionAuthHost)) // Session
+            {
+                proxyUrl = symConfig.sessionProxyURL;
+                proxyUserName = symConfig.sessionProxyUsername;
+                proxyPassword = symConfig.sessionProxyPassword;
+            }
+            else if (url.Contains(symConfig.keyAuthHost)) // Key Manager
+            {
+                proxyUrl = symConfig.keyProxyURL;
+                proxyUserName = symConfig.keyProxyUsername;
+                proxyPassword = symConfig.keyProxyPassword;
+            }
+            else if (url.Contains(symConfig.podHost)) // Pod
+            {
+                proxyUrl = symConfig.podProxyURL;
+                proxyUserName = symConfig.podProxyUsername;
+                proxyPassword = symConfig.podProxyPassword;
+            }
+            else if (url.Contains(symConfig.agentHost)) // Agent
+            {
+                proxyUrl = symConfig.agentProxyURL;
+                proxyUserName = symConfig.agentProxyUsername;
+                proxyPassword = symConfig.agentProxyPassword;
+            }
+
+            // Defaulting on global proxy
+            if (string.IsNullOrEmpty(proxyUrl))
+            {
+                proxyUrl = symConfig.proxyURL;
+                proxyUserName = symConfig.proxyUsername;
+                proxyPassword = symConfig.proxyPassword;
+            }
+
+            // Create and set proxy if any
+            if (!string.IsNullOrEmpty(proxyUrl))
+            {
+                var sessionProxy = new WebProxy();
+                var sessionProxyUri = new Uri(proxyUrl);
+                sessionProxy.Address = sessionProxyUri;
+                if (!string.IsNullOrEmpty(proxyUserName) && !string.IsNullOrEmpty(proxyPassword))
+                    sessionProxy.Credentials = new NetworkCredential(proxyUserName, proxyPassword);
+                else
+                    sessionProxy.Credentials = CredentialCache.DefaultNetworkCredentials; // Credential from current account
+                request.Proxy = sessionProxy;
+            }
+        }
+
         public string ReadResponse(HttpWebResponse resp)
         {
-            StreamReader reader = new StreamReader(resp.GetResponseStream(), Encoding.UTF8);
-            String responseString = reader.ReadToEnd();
+            var reader = new StreamReader(resp.GetResponseStream(), Encoding.UTF8);
```

Comment:
Please wrap in a "using" statement: https://docs.microsoft.com/en-us/dotnet/api/system.io.streamreader?view=netframework-4.8